mmv1/products/integrationconnectors/Connection.yaml

```diff
@@ -0,0 +1,930 @@
+# Copyright 2023 Google Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+---
+!ruby/object:Api::Resource
+name: "Connection"
+description: |
+  An Integration connectors Connection.
+references: !ruby/object:Api::Resource::ReferenceLinks
+  guides:
+    "Official Documentation": "https://cloud.google.com/integration-connectors/docs/createconnection"
+  api: "https://cloud.google.com/integration-connectors/docs/reference/rest/v1/projects.locations.connections"
+base_url: projects/{{project}}/locations/{{location}}/connections
+self_link: projects/{{project}}/locations/{{location}}/connections/{{name}}
+timeouts: !ruby/object:Api::Timeouts
+  insert_minutes: 30
+  update_minutes: 30
+  delete_minutes: 30
+custom_code: !ruby/object:Provider::Terraform::CustomCode
+  constants: templates/terraform/constants/integration_connectors_connection.go.erb
+  post_create: templates/terraform/post_create/integration_connectors_connection.go.erb
+create_url: projects/{{project}}/locations/{{location}}/connections?connectionId={{name}}
+update_verb: :PATCH
+update_mask: true
+autogen_async: true
```

Comment:
Can you confirm this again, I see that if I have this as is, it waits for LRO to be done and then also waits for state to be stable (not CREATING or UPDATING) which is the desired behaviour.

Consider a scenario where the update call is made, if disabled this, it might directly check for the state, this can happen before the state was even set to UPDATING.

Reply:
Also, the tests for update would fail, because, post_destroy doesn't wait for LRO to be completed, and it finds the resource and which fails the tests, so I would think we should still consider the LRO.

Reply:
I see. So we want LRO to be the first gate and state handler be the second to confirm if the operation is actually done?

Reply:
No, we want the state handler so that we can allow other operations on that resource.